src/main/java/bussiness_layer/services/impl/ProfessorService.java

```diff
@@ -79,11 +90,16 @@ public void updateLessons(String profUsername, List<LessonDto> lessonDtos) {
             lesson.setGrade(lessonDto.getGrade());
         });
         lessonRepository.flush();
-    }
-
-    private void notifyStudentsViaEmail() {
 
+        return lessonDtos.stream()
```

Comment:
I can't see where you check if the student wants to be notified or not through email.